examples/assets_pandas_type_metadata/assets_pandas_type_metadata/resources/csv_io_manager.py

```diff
@@ -4,26 +4,26 @@
 import pandas as pd
 from dagster import (
     AssetKey,
+    ConfigurableIOManager,
     MemoizableIOManager,
     TableSchemaMetadataValue,
-    io_manager,
 )
 from dagster._core.definitions.metadata import MetadataValue
+from pydantic import Field
 
 
-class LocalCsvIOManager(MemoizableIOManager):
+class LocalCsvIOManager(ConfigurableIOManager, MemoizableIOManager):
     """Translates between Pandas DataFrames and CSVs on the local filesystem."""
 
-    def __init__(self, base_dir):
-        self._base_dir = base_dir
+    base_dir: str = Field(default=os.getenv("DAGSTER_HOME"))
```

Comment:
I wasn't entirely sure what to do about `base_dir` for this conversion. Before it would always be `context.instance.storage_directory()` but we dont have access to the instance in the input/output contexts (as far as i know). Went with making it user config that defaults to DAGSTER_HOME but open to other ideas

Reply:
https://github.com/dagster-io/dagster/pull/13613 might help here, by making context available as a getter on the resource. Then, a cached property in the resource body can figure out which `base_dir` to use (user-supplied, if any, then from context)